### PR TITLE
azure: Remove hardcoded Container Linux version

### PR DESCRIPTION
**Pull request by justaugustus**
Currently, the Container Linux version is referenced in the `storage_image_reference` map for the creation of Azure resources as `version   = "${var.versions["container_linux"]}"`.

The `versions` var points to the `tectonic_versions` map, defined in `config.tf`.

The default map for `tectonic_versions` map hardcodes `container_linux`, which is causing Azure implementations to be instantiated w/ an old version of Container Linux.

This PR uses `version   = "latest"` instead, which will work across all Azure SKUs / `cl_channel`s.

@crawford: this fixes what we were discussing on the channel

cc: @alexsomesan @s-urbaniak @squat

### CODE REVIEW

**Comment by justaugustus** [comment]
Here's some output from a test cluster to show current CL version number is coming in okay and the cluster is healthy:

```
augustus@auggiebook:tectonic-installer$ kubectl get cs
NAME                 STATUS      MESSAGE                                                                                        ERROR
scheduler            Unhealthy   Get http://127.0.0.1:10251/healthz: dial tcp 127.0.0.1:10251: getsockopt: connection refused   
controller-manager   Unhealthy   Get http://127.0.0.1:10252/healthz: dial tcp 127.0.0.1:10252: getsockopt: connection refused   
etcd-2               Healthy     {"health": "true"}                                                                             
etcd-0               Healthy     {"health": "true"}                                                                             
etcd-1               Healthy     {"health": "true"}                                                                             
augustus@auggiebook:tectonic-installer$ kubectl get nodes -o wide
NAME                 STATUS    AGE       VERSION           EXTERNAL-IP   OS-IMAGE                                       KERNEL-VERSION
cl-ver-01-master-0   Ready     16m       v1.7.2+coreos.0   <none>        Container Linux by CoreOS 1409.8.0 (Ladybug)   4.11.12-coreos-r1
cl-ver-01-master-1   Ready     16m       v1.7.2+coreos.0   <none>        Container Linux by CoreOS 1409.8.0 (Ladybug)   4.11.12-coreos-r1
cl-ver-01-worker-0   Ready     16m       v1.7.2+coreos.0   <none>        Container Linux by CoreOS 1409.8.0 (Ladybug)   4.11.12-coreos-r1
cl-ver-01-worker-1   Ready     16m       v1.7.2+coreos.0   <none>        Container Linux by CoreOS 1409.8.0 (Ladybug)   4.11.12-coreos-r1
cl-ver-01-worker-2   Ready     16m       v1.7.2+coreos.0   <none>        Container Linux by CoreOS 1409.8.0 (Ladybug)   4.11.12-coreos-r1
augustus@auggiebook:tectonic-installer$ kubectl get deploy,ds,po -n tectonic-system
NAME                                           DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/container-linux-update-operator         1         1         1            1           14m
deploy/default-http-backend                    1         1         1            1           15m
deploy/kube-state-metrics                      1         1         1            1           14m
deploy/kube-version-operator                   1         1         1            1           14m
deploy/prometheus-operator                     1         1         1            1           15m
deploy/tectonic-channel-operator               1         1         1            1           14m
deploy/tectonic-console                        2         2         2            2           15m
deploy/tectonic-identity                       2         2         2            2           15m
deploy/tectonic-ingress-controller             1         1         1            1           15m
deploy/tectonic-monitoring-auth-alertmanager   1         1         1            1           14m
deploy/tectonic-monitoring-auth-prometheus     1         1         1            1           14m
deploy/tectonic-prometheus-operator            1         1         1            1           14m
deploy/tectonic-stats-emitter                  1         1         1            1           14m

NAME                                 DESIRED   CURRENT   READY     UP-TO-DATE   AVAILABLE   NODE-SELECTOR   AGE
ds/container-linux-update-agent-ds   5         5         5         5            5           <none>          14m
ds/node-agent                        5         5         5         5            5           <none>          14m
ds/node-exporter                     5         5         5         5            5           <none>          14m

NAME                                                       READY     STATUS    RESTARTS   AGE
po/alertmanager-main-0                                     2/2       Running   0          13m
po/alertmanager-main-1                                     2/2       Running   0          13m
po/container-linux-update-agent-ds-2nqvr                   1/1       Running   0          14m
po/container-linux-update-agent-ds-wxmqj                   1/1       Running   0          14m
po/container-linux-update-agent-ds-xs5bc                   1/1       Running   0          14m
po/container-linux-update-agent-ds-z3gx7                   1/1       Running   0          14m
po/container-linux-update-agent-ds-zz16v                   1/1       Running   0          14m
po/container-linux-update-operator-2925063197-n8ffc        1/1       Running   0          14m
po/default-http-backend-3970209879-vv6hm                   1/1       Running   0          15m
po/kube-state-metrics-3716310443-wdndl                     1/1       Running   0          14m
po/kube-version-operator-643279919-fm23f                   1/1       Running   0          14m
po/node-agent-9bsr3                                        1/1       Running   0          14m
po/node-agent-pk11r                                        1/1       Running   0          14m
po/node-agent-sbctk                                        1/1       Running   0          14m
po/node-agent-vr3s7                                        1/1       Running   0          14m
po/node-agent-z5fm6                                        1/1       Running   0          14m
po/node-exporter-0kddq                                     1/1       Running   0          14m
po/node-exporter-1mp0r                                     1/1       Running   0          14m
po/node-exporter-38mgm                                     1/1       Running   0          14m
po/node-exporter-dd6p4                                     1/1       Running   0          14m
po/node-exporter-l2jwd                                     1/1       Running   0          14m
po/prometheus-k8s-0                                        2/2       Running   0          14m
po/prometheus-operator-3133808683-bg0vm                    1/1       Running   0          15m
po/tectonic-channel-operator-3783788381-3h80w              1/1       Running   0          14m
po/tectonic-console-3448047066-7qp4r                       1/1       Running   0          15m
po/tectonic-console-3448047066-v9mgw                       1/1       Running   0          15m
po/tectonic-identity-1456953540-lmdtd                      1/1       Running   2          15m
po/tectonic-identity-1456953540-n9qs4                      1/1       Running   0          15m
po/tectonic-ingress-controller-2341783404-xcgfr            1/1       Running   0          15m
po/tectonic-monitoring-auth-alertmanager-389750782-l0xbh   1/1       Running   0          14m
po/tectonic-monitoring-auth-prometheus-1089940707-jg7wr    1/1       Running   0          14m
po/tectonic-prometheus-operator-2366017287-tfvcc           1/1       Running   0          14m
po/tectonic-stats-emitter-3536290556-djstw                 2/2       Running   0          14m
augustus@auggiebook:tectonic-installer$ kubectl logs po/container-linux-update-operator-2925063197-n8ffc -n tectonic-system
W0812 07:04:31.087237       1 main.go:42] Use of -manage-agent=true is deprecated and will be removed in the future
I0812 07:04:31.088650       1 main.go:53] /bin/update-operator running
I0812 07:04:31.263458       1 leaderelection.go:179] attempting to acquire leader lease...
I0812 07:04:31.290065       1 leaderelection.go:189] successfully acquired lease tectonic-system/container-linux-update-operator-lock
augustus@auggiebook:tectonic-installer$ kubectl logs po/container-linux-update-agent-ds-2nqvr -n tectonic-system
I0812 07:04:39.377807       1 main.go:42] /bin/update-agent running
I0812 07:04:39.377861       1 agent.go:79] Setting info labels
I0812 07:04:39.414957       1 agent.go:85] Marking node as schedulable
I0812 07:04:39.444450       1 agent.go:96] Setting annotations map[string]string{"container-linux-update.v1.coreos.com/reboot-in-progress":"false", "container-linux-update.v1.coreos.com/reboot-needed":"false"}
I0812 07:04:39.479941       1 agent.go:111] Waiting for ok-to-reboot from controller...
I0812 07:04:39.480648       1 agent.go:220] Beginning to watch update_engine status
I0812 07:04:39.481624       1 agent.go:175] Updating status
I0812 07:42:02.880234       1 agent.go:175] Updating status
I0812 07:42:12.916079       1 agent.go:175] Updating status
W0812 08:04:38.160004       1 agent.go:117] error waiting for an ok-to-reboot: waiting for annotation "container-linux-update.v1.coreos.com/reboot-ok" failed: watch closed before Until timeout
I0812 08:04:38.160020       1 agent.go:111] Waiting for ok-to-reboot from controller...
```

I'll leave this cluster up in case anyone needs to test anything internally. Just ping me for deets. :)

**Comment by justaugustus** [comment]
@crawford We do the same (search images based on `cl_channel` and then choose the most recent) for AWS: https://github.com/coreos/tectonic-installer/blob/master/modules/aws/master-asg/master.tf#L2

For bare metal, VMware, Openstack, we suggest downloading a recent version.

It just seems to me that something like ensuring backward compatibility would be something that happened in the CL release / promotion process.
Could you advise on a better approach for this PR?

**Comment by alexsomesan** [comment]
Been discussing options with @sym3tri on this and the intention was to still have this configurable, but default to latest.
Also, backporting this to the other platforms was part of the discussion.

IMO, it's worth having this configurable as an option.
Of course we can revert back to the old way, but the next time something bites in a new version of CL we'll be scrambling again for a way to contain it.

**Comment by squat** [comment]
From the email thread:
> Our plan (as discussed yesterday with the installer team & Rob/Reza) is to do a quick fix to pin to a newer version for the next dot release. Then we'll apply a proper fix in our next sprint for the September release.

We currently pin to the latest Container Linux version in AWS as well [0]. Let's merge this in now and add proper configurability for this feature after the release is cut.

[0] https://github.com/coreos/tectonic-installer/blob/master/modules/aws/master-asg/master.tf#L1-L2

**Comment by Quentin-M** [comment]
Thank you! I will cherry-pick that immediately and move forward with 1.7.1-tectonic.2

**Comment by crawford** [comment]
@justaugustus Yes, but AWS ASGs will continue booting that same version, effectively taking a snapshot (unless I am mistaken). Azure, on the other hand, actually has support for URN labels and will continue to actually boot the latest image (again, unless I am mistaken about how Terraform sets up Machine Scale Sets). The behavior between the two is distinctly different.

**Comment by justaugustus** [comment]
@crawford Excellent point. I hadn't thought of it that way.
Just took a peep at the AWS code and what it's doing is:
* Find the most recent AMI on `cl_channel` (https://github.com/coreos/tectonic-installer/blob/master/modules/aws/master-asg/master.tf#L2-L6)
* Creating a launch configuration based on that AMI ID
* Referencing that launch configuration in the ASG resource

Technically, it's doing the same as Azure i.e., finding a different image if the most recent has changed on subsequent `plan` / `apply`s

The key difference is a lifecycle hook in the AWS launch configuration (https://github.com/coreos/tectonic-installer/blob/master/modules/aws/master-asg/master.tf#L75), which prevents destroy / re-creation of the TF resources if / when the `image_id` changes.

**TL;DR**: you're totally right. We need a clever way of solving for this on the Azure side.

cc: @alexsomesan / @squat / @robszumski
